libbeat/publisher/queue/queue.go

```diff
@@ -34,6 +34,21 @@ type ACKListener interface {
 	OnACK(eventCount int) // number of consecutively published events acked by producers
 }
 
+//Metrics is a set of basic-user friendly metrics that report the current state of the queue. These metrics are meant to be relatively generic and high-level, and when reported directly, can be comprehensible to a user.
+type Metrics struct {
```

Comment:
This is a good point about what full means. It seems like just having a counter for events that were rejected because they wouldn't fit is what we'd want.

Reply:
Yah, that's a good point about "full" too. Perhaps leave that to the metrics code to derive,  and let this interface just report counts and byte values, as you suggested.

Reply:
I think that makes sense. The queue itself doesn't care if something can fit into it, that is a concern of whoever is using the Producer interface.